router/Cargo.toml

```diff
@@ -20,8 +20,8 @@ text-generation-client = { path = "client" }
 clap = { version = "^4.5.4", features = ["derive", "env"] }
 futures = "^0.3.30"
 flume = "^0.11.0"
-metrics = "0.21.1"
-metrics-exporter-prometheus = { version = "0.12.2", features = [] }
+metrics = "0.22.3"
+metrics-exporter-prometheus = { version = "0.14.0", features = [] }
```

Comment:
This change was required as well, the metrics endpoint just didn't work at all once the `metrics` crate was upgraded until this was upgraded too :/